Analytics/Classes/Internal/SEGSegmentIntegration.m

```diff
@@ -109,17 +109,44 @@ - (id)initWithAnalytics:(SEGAnalytics *)analytics httpClient:(SEGHTTPClient *)ht
         if ([NSThread isMainThread]) {
             [self setupFlushTimer];
         } else {
-            dispatch_sync(dispatch_get_main_queue(), ^{
+            dispatch_async(dispatch_get_main_queue(), ^{
                 [self setupFlushTimer];
             });
         }
+        
+        // Observe changes to flushInterval and update the timer
+        [self.configuration addObserver:self
```

Comment:
Figured we might want to observe this, let me know if you think this is overkill.